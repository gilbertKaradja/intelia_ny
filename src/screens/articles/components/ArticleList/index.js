import React, { Component } from 'react';
import ArticleEntry from './components/ArticleEntry/index.js';


import LoadingContent from './components/LoadingContent.js';
import ErrorLoadingContent from './components/ErrorLoadingContent.js';
import EmptyContent from './components/EmptyContent.js';

class ArticleList extends Component {

    render() {
        let { loading, errorLoading, data, activeArticleData } = this.props;
        let content;

        if (loading) {

            content = <LoadingContent />;

        } else if (!loading && errorLoading) {

            content = <ErrorLoadingContent retryButtonHandler={this.props.retryButtonHandler}/>;

        } else if (data && data.length > 0) {

            let itemRows = data.map(item => {
                return (
                    <ArticleEntry
                        key={item.id}
                        {...item}
                        onClickHandler={() => this.props.selectArticleHandler(item)}
                        selected={activeArticleData && activeArticleData.id === item.id}
                    />
                );
            });

            content = (
                <div className="articles_wrapper">
                    {itemRows}

                    <div className="load_control" onClick={this.props.loadMoreClickHandler}>
                        {this.props.fetchingMoreArticles && (
                            <div class="lds-ellipsis"><div></div><div></div><div></div><div></div></div>
                        )}

                        {!this.props.fetchingMoreArticles && (
                            <div className="text">Load More</div>
                        )}


                    </div>
                </div>
            );

        } else {
            content = <EmptyContent />;
        }

        return (
            <div className="ArticleList">
                {content}
            </div>
        );
    }
}

export default ArticleList;