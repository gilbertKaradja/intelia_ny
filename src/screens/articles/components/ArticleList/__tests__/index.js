import React from 'react';
import { shallow, mount } from 'enzyme';

import ArticleList from '../index.js';

let sampleOptions = [
    { label: 'Train', value: '1' },
    { label: 'Stop Sign', value: '9' },
    { label: 'Couch', value: 'joke' },
    { label: 'Random Sentence', value: 'evo993' },
    { label: 'Something', value: 'chihuahua' }
];

describe('ArticleList component', () => {

    it('should render a loading indicator when loading', () => {

        //renders initial dropdown properly
        const component = mount(
            <ArticleList loading={true} />
        );
        expect(component).toMatchSnapshot();

        component.unmount();
    });

    it('should render an error message and a retry button when failed to get posts', () => {

        const retryButtonHandler = jest.fn(value => value);

        //renders initial dropdown properly
        const component = mount(
            <ArticleList
                loading={false}
                errorLoading={true}
                retryButtonHandler={retryButtonHandler}
            />
        );
        expect(component).toMatchSnapshot();

        component.find('.retry_button').simulate('click');

        expect(retryButtonHandler.mock.calls.length).toBe(1);

        component.unmount();
    });

});