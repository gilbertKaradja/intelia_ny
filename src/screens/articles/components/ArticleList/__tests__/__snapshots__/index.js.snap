// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList component should render a loading indicator when loading 1`] = `
<ArticleList
  loading={true}
>
  <div
    className="ArticleList"
  >
    <LoadingContent>
      <div
        className="center_wrapper"
      >
        <div>
          <div
            className="loader_bars"
          >
            Loading...
          </div>
        </div>
      </div>
    </LoadingContent>
  </div>
</ArticleList>
`;

exports[`ArticleList component should render an error message and a retry button when failed to get posts 1`] = `
<ArticleList
  errorLoading={true}
  loading={false}
  retryButtonHandler={[MockFunction]}
>
  <div
    className="ArticleList"
  >
    <ErrorLoadingContent
      retryButtonHandler={[MockFunction]}
    >
      <div
        className="center_wrapper"
      >
        <div>
          <div
            className="description"
          >
            WE COULDN'T GET THE POSTS
          </div>
          <button
            className="retry_button"
            onClick={[MockFunction]}
          >
            Try Again
          </button>
        </div>
      </div>
    </ErrorLoadingContent>
  </div>
</ArticleList>
`;
